facebookresearch/ParlAI
# Now multiple checkpoints will be saved after using -stopk option in train_model.py

**Patch description**

In this PR, the [train_model.py](../blob/main/parlai/scripts/train_model.py) script has the option to keep and store multiple checkpoints based on validation metrics. Some checkpoints that have best validation metrics during training will be kept in the directory specified by `--model-file`. Resolves #4970 

**Testing steps**

To test this PR, you can run something like `parlai train_model --task babi:task10k:1 --model seq2seq --model-file seq2seq/babi_task10k --batchsize 32 --validation-every-n-secs 30 -stopk 5 -vstep 50`
Note that `-vstep` option must be exist to save and store 5 top checkpoints in this example. For me, the list of files in the `-model-file` directory after 1350 steps of training is like this:
![2023-03-10_16h39_48](https://user-images.githubusercontent.com/12621752/224324595-004160a1-2b0a-4083-9914-82aebc5e219b.png)
